src/inmanta/data/__init__.py

```diff
@@ -4519,6 +4519,30 @@ async def get_resources_in_latest_version(
                         result.append(resource)
         return result
 
+    @classmethod
+    async def get_resource_type_count_for_latest_version(cls, environment: uuid.UUID) -> dict[str, int]:
+        """
+        Returns the count for each resource_type over all resources in the model's latest version
+        """
+        query_latest_model = f"""
+            SELECT max(version)
+            FROM {ConfigurationModel.table_name()}
+            WHERE environment=$1
+        """
+        query = f"""
+            SELECT resource_type, count(*) as count
+            FROM {Resource.table_name()}
+            WHERE environment=$1 AND model=({query_latest_model})
+            GROUP BY resource_type;
+        """
+        values = [cls._get_value(environment)]
+        result: JsonType = {}
```

Comment:
Did you run mypy?

```suggestion
        result: dict[str, int] = {}
```

Reply:
That's weird, this wasn't caught by mypy. When I change this to `dict[str, int]` it does complain that `record["count"]` is an object and not an int though

Reply:
Ah. True. That's because `JsonType` matches `Dict[str, Any]`. The use of `Any` basically switches of type validation.

Reply:
I see, didn't know that !